src/actions/channels.ts

```diff
@@ -996,7 +996,7 @@ export function searchAllChannels(term: string, notAssociatedToGroup = '', exclu
             return {error};
         }
 
-        const channels = paginate ? response.channels : response;
+        const channels = (page == null && perPage == null) ? response : response.channels;
```

Comment:
Just wondering why we are testing against `null`. From my understanding those values will be either `undefined` or some number, is that right?  
Also, any reasons against testing the result (e.g. check if `response.channels` is defined) instead of the input?

Reply:
I think it's better to compare against the API response rather than the function parameter values. Thanks for the idea I have made the change.